packages/core-sdk/src/resources/dispute.ts

```diff
@@ -251,23 +278,194 @@ export class DisputeClient {
+                    type: "tuple",
+                  },
+                  {
+                    internalType: "address",
+                    name: "asserter",
+                    type: "address",
+                  },
+                  {
+                    internalType: "uint64",
+                    name: "assertionTime",
+                    type: "uint64",
+                  },
+                  {
+                    internalType: "bool",
+                    name: "settled",
+                    type: "bool",
+                  },
+                  {
+                    internalType: "contract IERC20",
+                    name: "currency",
+                    type: "address",
+                  },
+                  {
+                    internalType: "uint64",
+                    name: "expirationTime",
+                    type: "uint64",
+                  },
+                  {
+                    internalType: "bool",
+                    name: "settlementResolution",
+                    type: "bool",
+                  },
+                  {
+                    internalType: "bytes32",
+                    name: "domainId",
+                    type: "bytes32",
+                  },
+                  {
+                    internalType: "bytes32",
+                    name: "identifier",
+                    type: "bytes32",
+                  },
+                  {
+                    internalType: "uint256",
+                    name: "bond",
+                    type: "uint256",
+                  },
+                  {
+                    internalType: "address",
+                    name: "callbackRecipient",
+                    type: "address",
+                  },
+                  {
+                    internalType: "address",
+                    name: "disputer",
+                    type: "address",
+                  },
+                ],
+                internalType: "struct OptimisticOracleV3Interface.Assertion",
+                name: "",
+                type: "tuple",
+              },
+            ],
+            stateMutability: "view",
+            type: "function",
+          },
+        ],
+        functionName: "getAssertion",
+        args: [request.assertionId],
+      });
       const counterEvidenceHash = convertCIDtoHashIPFS(request.counterEvidenceCID);
       const encodedData = this.arbitrationPolicyUmaClient.disputeAssertionEncode({
         assertionId: request.assertionId,
         counterEvidenceHash,
       });
-      const txHash = await ipAccount.execute({
-        to: encodedData.to,
-        value: 0n,
-        data: encodedData.data,
-        operation: 0,
+      // Approve ipAccount to transfer wrappedIp, cannot use ipAccount.executeBatch because msg.sender is ipAccount with same address as spender.
+      await this.wrappedIpClient.approve({
+        spender: ipAccount.address,
+        amount: maxUint256,
       });
+      const contractCall = () =>
```

Comment:
all this new logic is only applicable if bond > 0. if bond === 0, lets skip all this and just call `disputeAssertionEncode` like before.